include/ChocAn/app/input_controller.hpp

```diff
@@ -20,25 +20,25 @@ Authors: Daniel Mendez
 #define CHOCAN_INPUT_CONTROLLER_H
 
 #include <map>
+#include <vector>
 #include <string>
 #include <functional>
 
 class Input_Controller
 {
 public:
 
-
-    using Form_Data   = std::map<std::string, std::string>;
-    using Field_Data  = std::pair<const std::string, std::string>;
-    using Form_Prompt = std::function<void(const std::string&)>;
+    using Fields         = std::vector<std::string>;
+    using Form_Data      = std::map<std::string, std::string>;
+    using Field_Callback = std::function<void(const std::string&)>;
 
     using Input_Control_Ptr = std::shared_ptr<Input_Controller>;
 
     // Read a single line of input
     virtual std::string read_input() const = 0;
 
     // Read a line of input for each field in fields
-    virtual Form_Data& read_form(Form_Data& fields, Form_Prompt prompt) = 0;
+    virtual Form_Data read_form(const Fields& fields, Field_Callback prompt) const = 0;
```

Comment:
read_form() method may be unnecessary. Will have to review.

Reply:
whether or not it needs to be virtual perhaps, but the function and implementation itself seems to solve its problem fairly well. i wouldn't change anything unless it ends up being a problem.